part/txn.go

```diff
@@ -349,20 +353,32 @@ func (txn *Txn[T]) delete(root *header[T], key []byte) (oldValue T, hadOld bool,
 			case parent.node.kind() == nodeKind256 && parent.node.size() <= 37:
 				newNode = (&node48[T]{header: *parent.node}).self()
 				newNode.setKind(nodeKind48)
-				children := newNode.node48().children[:0]
-				for _, n := range parent.node.node256().children[:] {
+				n48 := newNode.node48()
+				children := n48.children[:0]
+				for k, n := range parent.node.node256().children[:] {
 					if n != nil {
+						n48.index[k] = int8(len(children))
 						children = append(children, n)
 					}
 				}
 			case parent.node.kind() == nodeKind48 && parent.node.size() <= 12:
 				newNode = (&node16[T]{header: *parent.node}).self()
 				newNode.setKind(nodeKind16)
 				copy(newNode.children()[:], parent.node.children())
+				n16 := newNode.node16()
+				size := n16.size()
+				for i := 0; i < size; i++ {
+					n16.keys[i] = n16.children[i].prefix[0]
+				}
 			case parent.node.kind() == nodeKind16 && parent.node.size() <= 3:
 				newNode = (&node4[T]{header: *parent.node}).self()
 				newNode.setKind(nodeKind4)
 				copy(newNode.children()[:], parent.node.children())
+				n4 := newNode.node4()
+				size := n4.size()
+				for i := 0; i < size; i++ {
+					n4.keys[i] = n4.children[i].prefix[0]
```

Comment:
could we copy the n16 keys array instead here?

Reply:
oh yeah! leftover from when I hadn't converted node16 to this form yet.